src/backbone.marionette.js

```diff
@@ -1,6 +1,100 @@
-/*
-const Marionette = {
-};
+var Marionette;
+import extend             from './utils/extend';
+import isNodeAttached     from './utils/isNodeAttached';
+import mergeOptions       from './utils/mergeOptions';
+import getOption          from './utils/getOption';
+import proxyGetOption     from './utils/proxyGetOption';
+import normalizeMethods   from './utils/normalizeMethods';
+import normalizeUIString  from './utils/normalizeUIString';
+import normalizeUIKeys    from './utils/normalizeUIKeys';
+import normalizeUIValues  from './utils/normalizeUIValues';
+import deprecate          from './utils/deprecate';
+
+import VERSION            from './version';
+import MonitorDOMRefresh  from './dom-refresh';
+import MNObject           from './object';
+import Renderer           from './renderer';
+import TemplateCache      from './template-cache';
+import AbstractView       from './abstract-view';
+import View               from './view';
+import CollectionView     from './collection-view';
+import CompositeView      from './composite-view';
+import Behavior           from './behavior';
+import Behaviors          from './behaviors';
+import Region             from './region';
+import Application        from './application';
+import AppRouter          from './app-router';
+import MarionetteError    from './error';
+
+import {
+  FEATURES,
+  isEnabled,
+  setEnabled
+} from './features';
+
+import {
+  bindEntityEvents,
+  unbindEntityEvents,
+  proxyBindEntityEvents,
+  proxyUnbindEntityEvents
+} from './bind-entity-events';
+
+import {
+  proxyRadioHandlers,
+  unproxyRadioHandlers
+} from './radio-helpers';
+
+import {
+  triggerMethod,
+  triggerMethodOn,
+  triggerMethodMany
+} from './trigger-method';
+
+Marionette = Marionette || {};
+
+// extend Marionette in the meantime
+Object.assign(Marionette, {
```

Comment:
@jasonLaster If we're compiling ES6 modules (which aren't natively supported in any environment yet) we may as well compile Object.assign and any other `>ES5` features in the process. The global version of the library (a bundle compiled from an index file that imports everything, generating a `marionette.js` file that can be loaded via `<script>` tag) can contain the compiled code, so Object.assign will work anywhere. Some ES6+ features compile into nice clean ES5 code, while some others compile into complex, less-performant ES5 code (generator functions compiled for environments without native generator support incur a performance cost, for example (see [Facebook's regenerator](https://github.com/facebook/regenerator))). So, it would be a matter of choosing wisely what features to use and to compile at the same time as the ES6 modules for the global build.

For browserify users, the good news is that configuring all browserify transforms needed for the user's browserify project is in our hands, and the user need not worry about it.

In the case of Webpack, the app developer (not us) has to configure Webpack. The best we can do is clearly document which Webpack configurations are needed when and if the end user of the Marionette library wants to import ES6 code directly.

A solution to battle these things is to compile ES6 modules into a CommonJS format on `prepublish` so that the shipped library is consumable in all common build tools and environments, with all non ES5 things already pre-compiled.

To see an example of that, check this out (though the project is not complete): https://github.com/infamous/motor/blob/joe/package.json#L14

See this thread on details of how this package.json configuration is widely consumable in many build systems: http://forums.infamous.io/t/api-build-system-goals/46

There are still some things to do, which I've put off until the library itself becomes more mature (it's not at all ready for consumption yet): include the babel-runtime so that Babel's helper functions aren't duplicated in every built file, but are instead imported from `babel-runtime` as needed. Babel will place the proper require calls into the resulting CommonJS modules during compilation.